src/components/Dashboard/scenes/Emails/EmailHistory.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useEffect, useState, useCallback } from 'react';
+import axios from 'axios';
+import { EMAILAPI_URL } from '../../../../constants';
+import LoadingSVG from '../../../../assets/svg/LoadingSVG';
+import { type Email } from '../../../../types';
+
+const EmailHistory: React.FC<{ refreshCount: number }> = ({ refreshCount }) => {
+  const [emails, setEmails] = useState<Email[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const fetchEmails = useCallback(() => {
+    setIsLoading(true);
+    setTimeout(() => {
+      axios
+        .get(`${EMAILAPI_URL}/api/v1/sent`)
+        .then((response) => {
+          setEmails(response.data.emails);
+          setIsLoading(false);
+        })
+        .catch((error) => {
+          console.error('Error:', error);
+          setIsLoading(false);
+        });
+    }, 1000);
```

Comment:
Remove the timeout. Use a hook to sent mails

Reply:
```
const fetchEmails = useCallback(() => {
    setIsLoading(true);
    axios
        .get(`${EMAILAPI_URL}/api/v1/sent`)
        .then((response) => {
          setEmails(response.data.emails);
          setIsLoading(false);
        })
        .catch((error) => {
          console.error('Error:', error);
          setIsLoading(false);
        });
  }, []);

  useEffect(() => {
    fetchEmails();
  }, [fetchEmails]);
```

like this ?